db/db_impl/db_impl_open.cc

```diff
@@ -606,21 +623,24 @@ Status DBImpl::Recover(
     // updated to versions_->NewFileNumber() in RenameTempFileToOptionsFile.
     std::vector<std::string> file_names;
     if (s.ok()) {
-      s = env_->GetChildren(GetName(), &file_names);
+      if (dbname_ != immutable_db_options_.wal_dir) {
```

Comment:
This `if` can be combined with the `if` above.

Reply:
I may not be able to do this naturally. See comment in new commit.

Reply:
Here,
```
if best_efforts_recovery == true
  files_in_dbname can be used.
else if `dbname_ == wal_dir`
  files_in_wal_dir can be used.
else
  need to call GetChildren(dbname_).
```

Reply:
When comparing path, you may need to call `NormalizePath()`. Otherwise, `/path/to/file` will be treated as different from `/path/to//file`.

Reply:
I'm not sure if I need to call it on dbname_. I just called on each of them.